### PR TITLE
Remove default config from kubernetes resources

### PR DESCRIPTION
**Pull request by rossf7**
Some cleanup to remove `DefaultConfig` from the K8s resources (`namespace`, `service` and `endpoints`).

### CODE REVIEW

File: service/awsconfig/v5/resource/endpoints/resource.go [hunk 1]
```diff
@@ -46,6 +37,9 @@ type Resource struct {
 // New creates a new configured endpoints resource.
 func New(config Config) (*Resource, error) {
 	// Dependencies.
+	if reflect.DeepEqual(config.Clients, Clients{}) {
+		return nil, microerror.Maskf(invalidConfigError, "config.Clients must not be empty")
+	}
```
Comment (rossf7) [inline service/awsconfig/v5/resource/endpoints/resource.go]: I sneaked in checking the Clients dependency. Everything else is removing `DefaultConfig`.

Comment (xh3b4sd) [reply service/awsconfig/v5/resource/endpoints/resource.go]: We had this already somewhere. Anyway, I think a structure like `Clients` should provide its own `Validate` method in the long run and then we simply call this here.

Comment (rossf7) [reply service/awsconfig/v5/resource/endpoints/resource.go]: That is a nice approach. I don't have time to do it now but we can improve later.

Comment (xh3b4sd) [reply service/awsconfig/v5/resource/endpoints/resource.go]: Yeah yeah, for sure. All good. Just saying.

**Pull request merged**